charmhelpers/contrib/openstack/ha/utils.py

```diff
@@ -64,7 +84,79 @@ def update_dns_ha_resource_params(resources, resource_params,
     @param crm_ocf: Corosync Open Cluster Framework resource agent to use for
                     DNS HA
     """
+    _relation_data = {'resources': {}, 'resource_params': {}}
+    update_hacluster_dns_ha(charm_name(),
+                            _relation_data,
+                            crm_ocf)
+    resources.update(_relation_data['resources'])
+    resource_params.update(_relation_data['resource_params'])
+    relation_set(relation_id=relation_id, groups=_relation_data['groups'])
+
+
+def assert_charm_supports_dns_ha():
+    """Validate prerequisites for DNS HA
+    The MAAS client is only available on Xenial or greater
+    """
+    if lsb_release().get('DISTRIB_RELEASE') < '16.04':
+        msg = ('DNS HA is only supported on 16.04 and greater '
+               'versions of Ubuntu.')
+        status_set('blocked', msg)
+        raise DNSHAException(msg)
```

Comment:
More of a question, than an issue:  Is the `DNSHAException` supposed to be caught by the caller for it to do something with?  (If so, probably should add a `:raises DNSHAException: if release is < 16.04` to the docstring.

If not, doesn't this mean on a default Juju deploy that juju will just retry the failing hook?

Reply:
I think documenting the fact that this raises an exception is good anyway; I think that currently juju will just end up trying the failed hook execution which is not awesome - we don't have a nice way to handle this in non-reactive charms right now.